src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -0,0 +1,505 @@
+        """
+        operation_id = hook_ctx.operation_id
+        elements = self._await_elements(operation_id, response or requests.Response())
+        responses = self.partition_responses.get(operation_id)
+
+        if elements is None or responses is None:
+            return (response, error)
+
+        if len(responses) == 0:
+            if error is not None:
+                logger.error(error)
+            self._clear_operation(operation_id)
+            return (response, error)
+
+        updated_response = self._create_response(responses[0], elements)
+        self._clear_operation(operation_id)
+        return (updated_response, None)
+
+    def _get_pdf_pages(
+        self,
+        file_content: bytes,
+        split_size: int = 1,
+    ) -> Generator[Tuple[io.BytesIO, int, int], None, None]:
+        """Reads given bytes of a pdf file and split it into n file-like objects, each
+        with `split_size` pages.
+
+        Args:
+            file_content (bytes): Content of the PDF file.
+            split_size (int, optional): Split size, e.g. if the given file has 10 pages
+            and this value is set to 2 it will yield 5 documents, each containing 2 pages
+            of the original document. By default it will split each page to a separate file.
+
+        Yields:
+            Generator[Tuple[io.BytesIO, int, int], None, None]: Yield the file contents with
+            their page number and overall pages number of the original document.
+        """
+
+        pdf = PdfReader(io.BytesIO(file_content))
+        offset = 0
+        offset_end = len(pdf.pages)
+
+        while offset < offset_end:
+            new_pdf = PdfWriter()
+            pdf_buffer = io.BytesIO()
+
+            end = min(offset + split_size, offset_end)
+
+            for page in list(pdf.pages[offset:end]):
+                new_pdf.add_page(page)
+
+            new_pdf.write(pdf_buffer)
+            pdf_buffer.seek(0)
+
+            # 1-index the page numbers
+            yield pdf_buffer, offset + 1, offset_end
+            offset += split_size
+
+    def _parse_form_data(self, decoded_data: MultipartDecoder) -> FormData:
+        """
+        Parses the form data from the decoded multipart data.
+
+        Args:
+            decoded_data (MultipartDecoder): The decoded multipart data.
+
+        Returns:
+            FormData: The parsed form data.
+        """
+        form_data: FormData = {}
+
+        for part in decoded_data.parts:
+            content_disposition = part.headers[b"Content-Disposition"]
+            part_params = self._decode_content_disposition(content_disposition)
+            name = part_params.get("name")
+
+            if name is None:
+                continue
+
+            if name == PARTITION_FORM_FILES_KEY:
+                filename = part_params.get("filename") or SUBSTITUTE_FILENAME
+                form_data[PARTITION_FORM_FILES_KEY] = File(
+                    filename.split("/")[-1], part.content
```

Comment:
Does it assume that a filename will have at least one `/` inside? But user can pass a file with `name.pdf`

Reply:
Ah, unfortunately it's clever: https://www.joshwcomeau.com/career/clever-code-considered-harmful/#the-intern-test-1

Reply:
Just checked and there is actually one case when it could possibly be an empty list: `"".split()`.

Added a condition that will handle this edge case.